### ISSUE TITLE
bug: Query with optional `any` input still needs to be passed an input value

### ISSUE THREAD

**Issue by ssalka** [trpc/trpc#5769]
### Provide environment information

```
System:
  OS: Windows 10 10.0.19045
  CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  Memory: 4.17 GB / 15.90 GB
Binaries:
  Node: 20.10.0 - C:\Program Files\nodejs\node.EXE
  npm: 10.5.2 - C:\Program Files\nodejs\npm.CMD
  pnpm: 8.11.0 - ~\AppData\Local\pnpm\pnpm.EXE
  bun: 1.1.4 - ~\.bun\bin\bun.EXE
Browsers:
  Edge: Chromium (123.0.2420.97)
  Internet Explorer: 11.0.19041.4355
npmPackages:
  @trpc/client: 11.0.0-rc.374 => 11.0.0-rc.374+5027209bc
  @trpc/server: 11.0.0-rc.374 => 11.0.0-rc.374+5027209bc
  react: 19.0.0-rc-935180c7e0-20240524 => 19.0.0-rc-935180c7e0-20240524
  typescript: 5.4.5 => 5.4.5
```

### Describe the bug

When using `z.any().optional()` for an input type to a query, typechecking still wants you to pass a value. It is expected that the input could be omitted.

### Link to reproduction

https://stackblitz.com/edit/github-bsyd1n?file=src%2Fpages%2Findex.tsx&view=editor

### To reproduce

Open the given repro link to see side-by-side examples of `z.string().optional()` working as expected, but `z.any().optional()` requiring an input to be passed.

### Additional information

This was done using a client made from `createTRPCClient` rather than `createTRPCNext` since that is what I am using in my app.

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

## Funding
* You can sponsor this specific effort via a [Polar.sh](https://polar.sh) pledge below
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/trpc/trpc/issues/5769">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/trpc/trpc/issues/5769/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/trpc/trpc/issues/5769/pledge.svg">
</picture>
</a>

**Comment by KATT** [comment]
I have "fixed" it in #5774 but unsure if I should merge it

Current behavior matches the behavior of TypeScript

[Example](https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABMMAKGYAOIoC5ECGYAngJSIDeAvgLABQ99Kq5A9K4gDwC03igMuRA):

```ts
function fn(input: any) {}


fn() // <-- ❌
```

**Comment by KATT** [comment]
Yeah, we won't fix this in tRPC - it adds a lot of maintenance burden for something that works the way TypeScript works.

If you want to fix this, file/find an issue in TypeScript that should allow `type T =  any | undefined` to be anything else but `undefined`
